drivers/hmis/app/graphql/types/hmis_schema/client.rb

```diff
@@ -151,7 +151,13 @@ def external_ids
     end
 
     def enrollments(**args)
-      resolve_enrollments(**args)
+      # If current user has "detailed" access to any enrollment for this client, then we also resolve
+      # "limited access" enrollments (if permitted). The purpose is to show additional enrollment history
+      # for "my" clients, but not for other clients in the system that I can see.
+      # This would need to change if we wanted to support the ability to see limited enrollment details for all clients.
+      has_some_detailed_access = current_permission?(permission: :can_view_enrollment_details, entity: object)
+      scope = object.enrollments.viewable_by(current_user, include_limited_access_enrollments: has_some_detailed_access)
+      resolve_enrollments(scope, **args, dangerous_skip_permission_check: true)
```

Comment:
If the permissions can be checked on pre-loaded objects, you could possibly load all the enrollments for a given client (assuming that's the context) and preload the projects or whatever else you need to determine permission.  That might allow this to be only a handful of queries, but I agree, seems like this is moving toward a fairly heavy DB use, though maybe it's ok as long as we don't ever have hundreds of enrollments per client....

Reply:
Oh this only gets called once per client, not once per enrollment. It's "a lot of queries" (like maybe 10) just because our permission checks perform several queries, and we're doing 2 permission checks. This is the reason I had to bump the expected number of db queries in the test.